app/src/main/java/org/oppia/android/app/topic/lessons/StorySummaryViewModel.kt

```diff
@@ -6,8 +6,30 @@ import org.oppia.android.app.model.StorySummary
 /** [ViewModel] for displaying a story summary. */
 class StorySummaryViewModel(
   val storySummary: StorySummary,
-  private val storySummarySelector: StorySummarySelector
+  private val storySummarySelector: StorySummarySelector,
+  private val chapterSummarySelector: ChapterSummarySelector
 ) : TopicLessonsItemViewModel() {
+  private val arrayList = ArrayList<ChapterSummaryViewModel>()
+
+  val chapterSummaryItemList: List<ChapterSummaryViewModel> by lazy {
+    getRecyclerViewItemList()
+  }
+
+  private fun getRecyclerViewItemList(): ArrayList<ChapterSummaryViewModel> {
+    storySummary.chapterList.forEachIndexed { index, chapterSummary ->
+      arrayList.add(
+        ChapterSummaryViewModel(
+          chapterPlayState = chapterSummary.chapterPlayState,
+          explorationId = chapterSummary.explorationId,
+          chapterName = chapterSummary.name,
+          storyId = storySummary.storyId,
+          index = index,
+          chapterSummarySelector = chapterSummarySelector
+        )
+      )
+    }
+    return arrayList
+  }
```

Comment:
```suggestion
  private fun getRecyclerViewItemList(): List<ChapterSummaryViewModel> {
    return storySummary.chapterList.mapIndexed { index, chapterSummary ->
      ChapterSummaryViewModel(
        chapterPlayState = chapterSummary.chapterPlayState,
        explorationId = chapterSummary.explorationId,
        chapterName = chapterSummary.name,
        storyId = storySummary.storyId,
        index = index,
        chapterSummarySelector = chapterSummarySelector
      )
    }
  }
```

1. Prefer avoiding ArrayList (MutableList is the Kotlin idiomatic way to have modifiable lists)
2. Prefer to pass around List (since it's read-only and improves code robustness in the same way preferring val over var does)
3. Prefer to transform the list & compute a new one

Reply:
done